eknrenderer/eknrenderer/eknr-renderer.c

```diff
@@ -0,0 +1,1327 @@
+
+  /* Need to render each sub-template from here */
+  for (; *iter != NULL; ++iter)
+    {
+      uintmax_t rv = 0;
+      data->section_variable = *iter;
+
+      if (!mustache_render (api, userdata, token->section))
+        {
+          data->section_variable = last_section_variable;
+          return 0;
+        }
+    }
+
+  data->section_variable = last_section_variable;
+  return 1;
+}
+
+static uintmax_t
+_renderer_bool_sect_from_ht (mustache_api_t           *api,
+                             void                     *userdata,
+                             mustache_token_section_t *token,
+                             GVariant                 *variant)
+{
+  RendererMustacheData *data = userdata;
+
+  /* Need to render each sub-template from here */
+  if (g_variant_get_boolean (variant))
+    return mustache_render (api, userdata, token->section);
+
+  /* Nothing to do */
+  return 1;
+}
+
+static uintmax_t
+_renderer_str_sect_from_ht (mustache_api_t           *api,
+                            void                     *userdata,
+                            mustache_token_section_t *token,
+                            GVariant                 *variant)
+{
+  RendererMustacheData *data = userdata;
+  uintmax_t rv = 0;
+  const char *last_section_variable = data->section_variable;
+
+  data->section_variable = g_variant_get_string (variant, NULL);
+
+  /* Need to render each sub-template from here */
+  rv = mustache_render (api, userdata, token->section);
+
+  data->section_variable = last_section_variable;
+
+  return rv;
+}
+
+static uintmax_t
+_renderer_section_from_ht_variant (GVariant                 *variant,
+                                   mustache_api_t           *api,
+                                   void                     *userdata,
+                                   mustache_token_section_t *token)
+{
+  g_autofree char *msg = NULL;
+  const char *vt = g_variant_get_type_string (variant);
+
+  if (g_strcmp0 (vt, "as") == 0)
+    return _renderer_strv_sect_from_ht (api, userdata, token, variant);
+  else if (g_strcmp0 (vt, "b") == 0)
+    return _renderer_bool_sect_from_ht (api, userdata, token, variant);
+  else if (g_strcmp0 (vt, "s") == 0)
+    return _renderer_str_sect_from_ht (api, userdata, token, variant);
+
+  msg = g_strdup_printf ("No handler for section type %s on token %s", vt, token->name);
+  (*api->error) (api, userdata, __LINE__, msg);
+  return 1;
+}
+
+static uintmax_t
+_renderer_sect_from_ht (mustache_api_t           *api,
+                        void                     *userdata,
+                        mustache_token_section_t *token)
+{
+  RendererMustacheData *data = userdata;
+  g_autoptr(GVariant) value = g_variant_dict_lookup_value (data->variables,
+                                                           token->name,
+                                                           NULL);
+  g_autofree const char **strv = NULL;
```

Comment:
Unused?

Reply:
All three of them were in fact, removed.